spec/save_operation_callbacks_spec.cr

```diff
@@ -47,7 +48,16 @@ private class CallbacksSaveOperation < Post::SaveOperation
   end
 
   private def setup_required_attributes
-    title.value = "Title"
+    title.value = "Title" unless @skip_set_required
+  end
+end
+
+private class SaveLineItem < LineItem::SaveOperation
+  permit_columns :name
+  before_save prepare
+
+  def prepare
+    true
```

Comment:
I'm not sure I understand what this is testing. Maybe there could be an an instance var that changes from false to true that you can assert in the specs? `property prepared = false` then set that to true to ensure it runs?

Reply:
Before this PR, calling this `before_save prepare` wouldn't call the `before_save set_uuid`. Without this `before_save prepare` it does call that parent one. So I just needed a before_save at this level. It doesn't really matter what it does. So before this PR, it would fail, but now it doesn't.

I'm open to options to make it more readable, but I just need a before_save that does anything here for the test.

Reply:
I think what worries me is that you could remove the `before_save prepare` and the specs would all still pass. So I think this prepare should set some variable  we could test, or add a comment explaining what it is testing.  What do you think?

Reply:
Ah, I see what you're saying. Ok, I can do that

Reply:
Ok, so the spec as it was no longer made sense once the require validations moved. Now I've added a new subclass to show that the `before_save` actually trickle down from the top with using some properties.